corehq/project_limits/rate_limiter.py

```diff
@@ -235,3 +248,23 @@ def _get_rate_definition_dict(rate_definition):
         attribute.name: getattr(rate_definition, attribute.name)
         for attribute in RateDefinition.__attrs_attrs__
     }
+
+
+class GaugeLimiter:
```

Comment:
There's a disconnect for me between the genericness of the names `Gauge` and `GuageLimiter` and the specificness (to measuring pillow offsets) of the implementation.

`RateLimiter` is set up generically versus how it is used to implement a specific measurement like the rates of form submissions (https://github.com/dimagi/commcare-hq/blob/1c970475c96197a30f2f850344927adda229160b/corehq/apps/receiverwrapper/rate_limiter.py#L56), and emulating that organization.

Alternatively, if you think we won't have another use case for tracking an evolving value over time like this for the purposes of making throttling decisions, and you think it's not worth making generic, then perhaps you can rename these classes to make clear that they're specific to tracking pillow lag.

I think I prefer the generic approach, because it separates logically generic behavior (defined in the class definition) from the more complex context-specific functions for pulling data etc (passed in when the class is instantiated).

Reply:
Thanks for bringing it up, Danny. I see your point. I started it out as a generic implementation but over the time it became very much specific to the pillows.
I will try to break up the classes to have a generic one and the one specific for the pillows.